File: Data Structures & Algorithms/three-integer-sum/submission-0.java
class Solution {
    public List<List<Integer>> threeSum(int[] nums) {
        Set<List<Integer>> set = new HashSet<>();
        Arrays.sort(nums);
        int L = 0;
        int R = nums.length - 1;

        for (int i = 0; i < nums.length; i++) {
            List<Integer> temp = new ArrayList<>();
            // we don't want to use the same starting element twice
            // this check works due to the array being sorted
            if ((i > 0) && (nums[i] == nums[i - 1])) {
                continue;
            }
            int a = nums[i];
            while (L < R) {
                if ((nums[L] + nums[R] + a) > target) {
                    R--;
                } else if ((nums[L] + nums[R] + a) < target) {
                    L++;
                } else {
                    temp.add(a);
                    temp.add(nums[L]);
                    temp.add(nums[R]);
                }
            }
            set.add(temp);
        }
        // make sure to convert the Set to List, such that we match
        // the given return type
        return new ArrayList<>(set);
    }
}

File: Data Structures & Algorithms/contains-duplicate-ii/submission-0.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        for i in range(len(nums)):
            for j in range(i + 1, len(nums)):
                if nums[i] == nums[j] and abs(i - j) <= k:
                    return True
        return False

"""
- given an integer array nums and integer k, return true if there are 2 distinct indices i
and j in the array such that nums[i] == nums[j] and abs(i - j) <= k. Otherwise return false.
- 
"""